code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -476,13 +476,35 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	description = "100 proof cinnamon schnapps, made for alcoholic teen girls on spring break."
 	color = "#FFFF91" // rgb: 255, 255, 145
 	boozepwr = 25
-	quality = DRINK_VERYGOOD
+	quality = DRINK_NICE  // teenager girls think it's DRINK_VERYGOOD
 	taste_description = "burning cinnamon"
 	glass_icon_state = "goldschlagerglass"
 	glass_name = "glass of goldschlager"
 	glass_desc = "100% proof that teen girls will drink anything with gold in it."
 	shot_glass_icon_state = "shotglassgold"
 
+/datum/reagent/consumable/ethanol/goldschlager/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume)
+	// Reset quality if someone's sharing the bottle
+	quality = initial(quality)
```

Comment:
Add a reagent-specific variable with the new quality if the conditions meet instead.
Something like
```
/// This is really popular with a certain demographic.
var/teenage_quality = DRINK_VERYGOOD
```

Reply:
But I already did this on line 490?